engine/step/step.go

```diff
@@ -493,11 +495,16 @@ func AfterRun(st *Step, values *values.Values, ss StateSetter) {
 				break
 			}
 		}
-		for step, state := range sc.Then {
-			if step == stepRefThis {
-				step = st.Name
+		switch sc.Type {
+		case condition.LOOP:
+			ss(st.Name, StateInLoop, sc.Message)
```

Comment:
`sc.Message` seems inappropriate for me in this case. During the execution, the state of the step could contains the current loop iteration number, maybe?

Reply:
Do you mean something like this?
```go
if st.MaxRetries < 0 || st.TryCount <= st.MaxRetries {
	state := fmt.Sprintf("%s#%d", StateLooping, st.TryCount)
	ss(st.Name, state, "")
} else {
	ss(st.Name, StateDone, "")
	logrus.Debugf("AfterRun: Step [%s] loop condition: max_retries (%d) exceeded", st.Name, st.MaxRetries)
}
```